### PR TITLE
add install directives in cmake

### PR DESCRIPTION
**Pull request by stracedude**
## Description

Add install directives in cmake files.

Hierarchy:
```
├── bin
│   └── walletconsole
├── include
│   ├── protobuf
│   ├── TrezorCrypto
│   ├── TrustWalletCore
│   └── WalletCore
└── lib
    ├── libprotobuf.a
    ├── libTrezorCrypto.a
    └── libTrustWalletCore.a

```

TrustWalletCore, TrezorCrypto headers directory for fully using library from cpp project.

## Testing instructions

make DESTDIR=${PWD} install
find usr/local/

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -97,3 +99,14 @@ if(NOT ANDROID AND NOT IOS_PLATFORM)
 endif()
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/swift/cpp.xcconfig.in ${CMAKE_CURRENT_SOURCE_DIR}/swift/cpp.xcconfig @ONLY)
+
+install(TARGETS TrustWalletCore
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        )
+
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/WalletCore
```
Comment (hewigovens) [inline CMakeLists.txt:108]: you don't need to use those C++ headers?

**Comment by hewigovens** [comment]
There is one more lib to install: `libprotobuf.a`

**Comment by catenocrypt** [comment]
Looks good.  I'll test if it is ok within the Docker image as well.

**Comment by stracedude** [comment]
> There is one more lib to install: `libprotobuf.a`

Protobuf is an external dependency, I don't think it needs to be installed. 

And after #977, it doesn't make sense.

File: CMakeLists.txt (hunk 1 above)
Comment (stracedude) [reply CMakeLists.txt:108]: I use the library in the cpp project, so using C API is less convenient, also overhead when converting from json -> protobuf -> cpp object.

Also, C API does not provide the full functionality of the library, for example, to get a list of selected unspent transactions from the plan, I wanted to create an issue, but I didn't get around.

**Comment by hewigovens** [comment]
@stracedude `libTrustWalletCore.a` depends on `libprotobuf.a` and `libTrezorCrypto.a`, we should install it for completeness, so user can either uses C interface + proto models or C++ interface without installing other dev libraries

**Review by hewigovens — APPROVED** [review]

**Comment by catenocrypt** [comment]
`make install` works inside Docker environment as well, installs under `/usr/local`.

**Review by catenocrypt — APPROVED** [review]

**Pull request merged**